src/controllers/AuthController.ts

```diff
@@ -0,0 +1,56 @@
+import API, { AuthAPI, SigninData, SignupData } from "../api/AuthAPI";
+import store from "../utils/Store";
+import router from "../utils/Router";
+import MessagesController from "./MessagesController";
+
+export class AuthController {
+  private readonly api: AuthAPI;
+
+  constructor() {
+    this.api = API;
```

Comment:
Лучше написать вот так
```suggestion
    this.api = new API();
```

или что-то типа того. Чтобы класс сам контролировал создание своих зависимостей

Reply:
Не уверен, что это правильно. Но в AuthAPI написано: export default new AuthAPI();

Reply:
Я понимаю, но сейчас класс ```AuthController``` использует объект, который создается где-то вовне. Если этот объект. Если этот объект удалить из памяти, то и работа ```AuthController``` будет нарушена. 
Я так понимаю, что ты хотел сделать что-то типа синглтона, но сейчас это не синглтон. Сейчас просто экспортируется один инстанс объекта. Синглтон реализуется через статический метод и при любом создании объекта будет будет возвращаться один и тот же инстанс. Про синглтон можешь почитать [тут](https://refactoring.guru/design-patterns/singleton/typescript/example)